### PR TITLE
Extensions: Add Advanced Tab settings for WP Super Cache

### PR DESCRIPTION
**Pull request by donnapep**
This PR adds settings for the Advanced tab of WP Super Cache. For the most part, the settings page is similar to the WordPress plugin version, with the following exceptions:

- The 'Caching On' and 'Caching Off' radio buttons have been consolidated into a single toggle switch.
- Checkboxes have been changed to toggle switches.
- Save button text was changed to 'Save Settings'.
- Extraneous text has not been carried over (e.g. the notes on garbage collection, Mod Rewrite Rules etc.).
- The UI better aligns with other Calypso settings pages.

[Calypso.live link](https://calypso.live/extensions/wp-super-cache?branch=add/wpsc-advanced-settings)

Below are screenshots of the individual sections. Note that hardcoded data is being used until the REST API is ready.

_Caching_

caching

_Miscellaneous_

miscellaneous

_Advanced_

advanced

_Cache Location_

cache location

_Expiry Time & Garbage Collection_

expiry time

_Accepted Filenames & Rejected URIs_

accepted filenames

_Rejected User Agents_

rejected user agents

_Lock Down_

lock down

_Directly Cached Files_

directly cached files

_Fix Configuration_

fix configuration

### CODE REVIEW

File: client/extensions/wp-super-cache/components/advanced/expiry-time.jsx [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+				<FormSelect
+					id="cache_schedule_interval"
+					name="cache_schedule_interval"
+					onChange={ handleSelect }
+					value={ fields.cache_schedule_interval || 'five_minutes_interval' }>
+					<option value="five_minutes_interval">{ translate( 'Once every five minutes' ) }</option>
+					<option value="jetpack_sync_interval">{ translate( 'Every 5 minutes' ) }</option>
+					<option value="minutes_10">{ translate( 'Every 10 minutes' ) }</option>
+					<option value="minutes_30">{ translate( 'Every 30 minutes' ) }</option>
```
Comment (a8ci18n) [inline client/extensions/wp-super-cache/components/advanced/expiry-time.jsx]: Hi! I've found a possible matching string that has already been translated **28** times:
`translate( '30 Minutes' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/wp-super-cache/components/advanced/expiry-time.jsx [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+					{ translate( 'Interval' ) }
+				</FormLabel>
+
+				<FormSelect
+					id="cache_schedule_interval"
+					name="cache_schedule_interval"
+					onChange={ handleSelect }
+					value={ fields.cache_schedule_interval || 'five_minutes_interval' }>
+					<option value="five_minutes_interval">{ translate( 'Once every five minutes' ) }</option>
+					<option value="jetpack_sync_interval">{ translate( 'Every 5 minutes' ) }</option>
```
Comment (a8ci18n) [inline client/extensions/wp-super-cache/components/advanced/expiry-time.jsx]: Hi! I've found a possible matching string that has already been translated **7** times:
`translate( '5 Minutes' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/wp-super-cache/components/advanced/expiry-time.jsx [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+						name="cache_schedule_type"
+						onChange={ handleRadio }
+						value="time" />
+					<span>
+						{ translate( 'Clock' ) }
+						<FormTextInput
+							disabled={ 'time' !== fields.cache_schedule_type }
+							onChange={ handleChange( 'cache_scheduled_time' ) }
+							value={ fields.cache_scheduled_time || '' } />
+						{ translate( 'HH:MM' ) }
```
Comment (a8ci18n) [inline client/extensions/wp-super-cache/components/advanced/expiry-time.jsx]: Hi! I've found a possible matching string that has already been translated **33** times:
`translate( 'HH:mm', { context: 'momentjs format string (for LT)'} )` *ES Score: 22*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by yoavf** [comment]
We'll want to import the wpsc strings from [t.wp.org](https://translate.wordpress.org/projects/wp-plugins/wp-super-cache) once this lands. @donnapep  can you please add the " [Status] Needs i18n Review" label before merging? thanks!

File: client/extensions/wp-super-cache/components/advanced/expiry-time.jsx [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+				<FormSelect
+					id="cache_schedule_interval"
+					disabled={ 'time' !== cache_schedule_type }
+					name="cache_schedule_interval"
+					onChange={ handleSelect }
+					value={ cache_schedule_interval || 'five_minutes_interval' }>
+					<option value="five_minutes_interval">{ translate( 'Once every five minutes' ) }</option>
+					<option value="jetpack_sync_interval">{ translate( 'Every 5 minutes' ) }</option>
+					<option value="minutes_10">{ translate( 'Every 10 minutes' ) }</option>
+					<option value="minutes_30">{ translate( 'Every 30 minutes' ) }</option>
```
Comment (a8ci18n) [inline client/extensions/wp-super-cache/components/advanced/expiry-time.jsx]: Hi! I've found a possible matching string that has already been translated **28** times:
`translate( '30 Minutes' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/wp-super-cache/components/advanced/expiry-time.jsx [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+				</FormLabel>
+
+				<FormSelect
+					id="cache_schedule_interval"
+					disabled={ 'time' !== cache_schedule_type }
+					name="cache_schedule_interval"
+					onChange={ handleSelect }
+					value={ cache_schedule_interval || 'five_minutes_interval' }>
+					<option value="five_minutes_interval">{ translate( 'Once every five minutes' ) }</option>
+					<option value="jetpack_sync_interval">{ translate( 'Every 5 minutes' ) }</option>
```
Comment (a8ci18n) [inline client/extensions/wp-super-cache/components/advanced/expiry-time.jsx]: Hi! I've found a possible matching string that has already been translated **7** times:
`translate( '5 Minutes' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/wp-super-cache/components/advanced/expiry-time.jsx [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+						name="cache_schedule_type"
+						onChange={ handleRadio }
+						value="time" />
+					<span>
+						{ translate( 'Clock' ) }
+						<FormTextInput
+							disabled={ 'time' !== cache_schedule_type }
+							onChange={ handleChange( 'cache_scheduled_time' ) }
+							value={ cache_scheduled_time || '' } />
+						{ translate( 'HH:MM' ) }
```
Comment (a8ci18n) [inline client/extensions/wp-super-cache/components/advanced/expiry-time.jsx]: Hi! I've found a possible matching string that has already been translated **33** times:
`translate( 'HH:mm', { context: 'momentjs format string (for LT)'} )` *ES Score: 22*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by donnapep** [comment]
@donnchawp I expect you'll be especially interested in [this bit](https://github.com/Automattic/wp-calypso/blob/e0eec8e49e17a4e0fc573b5676632b6bea4de753/client/extensions/wp-super-cache/components/wrap-settings-form.jsx#L60) where it lists out each section and which settings are used by those sections. I would imagine we'll need to discuss to ensure naming and structure are consistent.

File: client/extensions/wp-super-cache/components/advanced/expiry-time.jsx [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+					<FormSelect
+						id="cache_schedule_interval"
+						disabled={ 'time' !== cache_schedule_type }
+						name="cache_schedule_interval"
+						onChange={ handleSelect }
+						value={ cache_schedule_interval || 'five_minutes_interval' }>
+						<option value="five_minutes_interval">{ translate( 'Once every five minutes' ) }</option>
+						<option value="jetpack_sync_interval">{ translate( 'Every 5 minutes' ) }</option>
+						<option value="minutes_10">{ translate( 'Every 10 minutes' ) }</option>
+						<option value="minutes_30">{ translate( 'Every 30 minutes' ) }</option>
```
Comment (a8ci18n) [inline client/extensions/wp-super-cache/components/advanced/expiry-time.jsx]: Hi! I've found a possible matching string that has already been translated **28** times:
`translate( '30 Minutes' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/wp-super-cache/components/advanced/expiry-time.jsx [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+					</FormLabel>
+
+					<FormSelect
+						id="cache_schedule_interval"
+						disabled={ 'time' !== cache_schedule_type }
+						name="cache_schedule_interval"
+						onChange={ handleSelect }
+						value={ cache_schedule_interval || 'five_minutes_interval' }>
+						<option value="five_minutes_interval">{ translate( 'Once every five minutes' ) }</option>
+						<option value="jetpack_sync_interval">{ translate( 'Every 5 minutes' ) }</option>
```
Comment (a8ci18n) [inline client/extensions/wp-super-cache/components/advanced/expiry-time.jsx]: Hi! I've found a possible matching string that has already been translated **7** times:
`translate( '5 Minutes' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/wp-super-cache/miscellaneous.jsx [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+							</span>
+						</FormToggle>
+					</FormFieldset>
+				</form>
+			</Card>
+		</div>
+	);
+};
+
+const getFormSettings = settings => {
```
Comment (pgk) [inline client/extensions/wp-super-cache/miscellaneous.jsx]: I see this pattern in many of the files: a function we pass into `WrapSettingsForm` that returns a subset of settings fields. Was wondering if we should make this more declarative, by just passing the string array. 

Although passing in a function is more flexible.

Comment (donnapep) [inline client/extensions/wp-super-cache/miscellaneous.jsx]: Yeah, that would make sense to do as well. There's also [this](https://github.com/Automattic/wp-calypso/pull/12380/files#diff-f8410bc52a466929f7d47d48dd190e00R181) where more logic is needed to determine the contents of the array. It's conceivable that in the future other components might need to do something similar, hence your comment about flexibility. 😄

**Comment by rickybanister** [comment]
This got merged before any kind of design review—I know it's not public-facing as yet, but should we take a run-through and open a big issue with any things we find? 

In following along I just noticed a few small bits like indentations of italic description text and whatnot.

**Comment by donnapep** [comment]
@rickybanister I'm not sure we'll be doing design reviews in this early stage. We just want to get something in for now and then iterate later, which will likely completely change what we have. @dbtlr What are your thoughts on this?

**Comment by dbtlr** [comment]
I'd say feel free to nitpick some of the details, but let's stay away from big refreshes for now. Hopefully, if we can collect enough of the smaller things into a task list, we can do it before this goes live as a final cleanup task. For now, though, the goal is to get this in there and functioning, so let's not get too caught up in fixing them just yet.

**Comment by donnapep** [comment]
@rickybanister Given the above comments, how would you like to handle this going forward?

**Comment by rickybanister** [comment]
I totally understand. I know we have to make progress here and just get things converted over, but I also want to be respectful to Calypso's contributing guidelines and design patterns. If we adhere to our own rules it will make it easier for other plugin developers to follow those as well. 

I'm happy to either open a single issue highlighting a few small things to adjust, or we can hop on a call and share the screen and just click through, pointing out/recording small discrepancies as we find them. Let me know which you prefer.

**Comment by donnapep** [comment]
@rickybanister I'm going to be at a meetup this week, so if you could open an issue, that would be fantastic! There's also PR #12647 and #12689 if you'd like to have a look at those as well. Even if I don't get around to addressing the issues right away, at least I can avoid the same mistakes on the remaining tabs. Thx!

**Pull request merged**